icinga-app/icinga2.cmake

```diff
@@ -1,29 +1,46 @@
 #!/bin/sh
 ICINGA2_BIN=@CMAKE_INSTALL_FULL_LIBDIR@/icinga2/sbin/icinga2
 
-if test "x`uname -s`" = "xLinux" -a "x$1" = "xconsole"; then
-  libedit_line=`ldd $ICINGA2_BIN 2>&1 | grep libedit`
-  if test $? -eq 0; then
-    libedit_path=`echo $libedit_line | cut -f3 -d' '`
-    if test -n "$libedit_path"; then
-      libdir=`dirname -- $libedit_path`
-      found=0
-      for libreadline_path in `ls -1r -- $libdir/libreadline.so.* 2>/dev/null`; do
-        found=1
-        break
-      done
-      if test $found -eq 0; then
-        libdir2=/`echo $libdir | cut -f3- -d/`
-        for libreadline_path in `ls -1r -- $libdir2/libreadline.so.* 2>/dev/null`; do
+if test "x`uname -s`" = "xLinux"; then
+  if test "x$1" = "xconsole"; then
+    libedit_line=`ldd $ICINGA2_BIN 2>&1 | grep libedit`
+    if test $? -eq 0; then
+      libedit_path=`echo $libedit_line | cut -f3 -d' '`
+      if test -n "$libedit_path"; then
+        libdir=`dirname -- $libedit_path`
+        found=0
+        for libreadline_path in `ls -1r -- $libdir/libreadline.so.* 2>/dev/null`; do
           found=1
           break
         done
-      fi
-      if test $found -gt 0; then
-        export LD_PRELOAD="$libreadline_path:$LD_PRELOAD"
+        if test $found -eq 0; then
+          libdir2=/`echo $libdir | cut -f3- -d/`
+          for libreadline_path in `ls -1r -- $libdir2/libreadline.so.* 2>/dev/null`; do
+            found=1
+            break
+          done
+        fi
+        if test $found -gt 0; then
+          LD_PRELOAD="$libreadline_path:$LD_PRELOAD"
+        fi
       fi
     fi
   fi
+
+  for lib in /usr/lib{,64,/{{i386,x86_64}-linux-gnu,arm-linux-gnueabihf}}/libjemalloc.so.{1,2}; do
```

Comment:
Is there any system where the lib we and up using is located somewhere else than `@CMAKE_INSTALL_FULL_LIBDIR@`? If not, we could avoid all this guessing.

Also, this would prefer `libjemalloc.so.1` over `.2` (if we would break, see next comment, not sure what happens if you just write both to `LD_PRELOAD`). So maybe do something like above with `ls -1r`, though this would break with `.9` and `.10` and with spaces in the path, so not ideal as well, unfortunately.

Reply:
The mentioned variable tells where Icinga is installed to (user-definable), but the paths above locate libjemalloc (predefined by packages).

Reply:
Yes, but shouldn't this be the same for all the packages we build?